search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/DefaultLlmImpl.java

```diff
@@ -105,7 +105,9 @@ protected Map<String, String> getInputParameters(ChatCompletionInput chatComplet
                 );
             inputParameters.put(CONNECTOR_INPUT_PARAMETER_MESSAGES, messages);
             // log.info("Messages to LLM: {}", messages);
-        } else if (chatCompletionInput.getModelProvider() == ModelProvider.BEDROCK) {
+        } else if (chatCompletionInput.getModelProvider() == ModelProvider.BEDROCK
+            || chatCompletionInput.getModelProvider() == ModelProvider.COHERE
+            || chatCompletionInput.getLlmResponseField() != null) {
```

Comment:
Seems like we need to update the else block accordingly.

Reply:
You mean we need to say something about llm_response_field?

Reply:
Yeah that's what I was wondering.

Reply:
I updated it.